### ISSUE TITLE
Low FPS

### ISSUE THREAD

**Issue by Zybyte85** [foxnne/pixi#67]
When I am in the editor, I an getting about 10 FPS. I am on Fedora Linux and I compiled the program from the latest source, so that may be the case.

**Comment by foxnne** [comment]
Interesting, I haven't encountered this on either of my machines (macOS/Windows). Is the fps low before you open a project folder, just on the empty start page? I don't currently have a Linux machine, I can try WSL and see if I can replicate the issue maybe? This is the first report I've had of low fps.

There are certain things that are poorly optimized right now, but I've yet to have it be an issue. One of these things is that I am recursively checking the filesystem from the root project folder down, which means a very large file structure would get slow quickly. The other thing that comes to mind is when drawing with a very large sized brush, I noticed some FPS drops.

Totally understand if you aren't looking to spend time on it, but if you are, if you could give any more information on where/when you're experiencing low fps or if its the entire time the program is running that would be very helpful.

**Comment by Zybyte85** [comment]
Yeah it is only when I am in the actual editor itself and a file is open. When a file is not open, it gets 30 fps. The main thing I could think of is it being the Mesa drivers Linux uses. CPU and ram usage are normal, nothing really spikes up that much.  I also have the terminal output below if that helps.

`info(mach): found Vulkan backend on Integrated GPU adapter: Intel(R) UHD Graphics 630 (CFL GT2), Intel open-source Mesa driver: Mesa 24.1.7`

**Comment by foxnne** [comment]
There are a ton of opportunities to optimize things in Pixi, several of which I'm keen to hopefully start looking into soon. I'm sorry the performance is poor on your machine, hopefully I can improve it in the future. The other thing you may try in the meantime is compiling in ReleaseFast mode, as debug builds are often a lot slower.

`zig build run -Doptimize=ReleaseFast`

**Comment by Zybyte85** [comment]
Yeah that worked! It was running at 120 fps in the menu there for a second lol. You might want to put a note of that in the readme.

**Comment by foxnne** [comment]
Glad that worked! The only reason I didn't make it default yet is feeling hesitant, I wanted to try to learn how to use the debug build to output the log when the program crashes, but I haven't looked into it yet.
